var/spack/repos/builtin/packages/intel-mkl/package.py

```diff
@@ -142,6 +142,8 @@ def install(self, spec, prefix):
     def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
         # set up MKLROOT for everyone using MKL package
         spack_env.set('MKLROOT', self.prefix)
+        spack_env.set('SPACK_COMPILER_EXTRA_RPATHS',
```

Comment:
Sorry, missed the OZp description. I think u should add to this env variable (in case it's set by someone else)